test/suites/integration/ram_share_test.go

```diff
@@ -0,0 +1,352 @@
+			g.Expect(gw.Status.Conditions[1].Status).To(BeEquivalentTo(metav1.ConditionFalse))
+			g.Expect(gw.Status.Conditions[1].Reason).To(BeEquivalentTo(gwv1.GatewayReasonInvalid))
+		}).Should(Succeed())
+	})
+
+	AfterEach(func() {
+		// Delete gateways in test namespace except the framework gateway
+		gws := &gwv1.GatewayList{}
+		err := testFramework.Client.List(ctx, gws, client.InNamespace(k8snamespace))
+		Expect(err).To(BeNil())
+		for _, gw := range gws.Items {
+			if gw.Name != testGateway.Name {
+				testFramework.ExpectDeletedThenNotFound(ctx, &gw)
+			}
+		}
+	})
+
+	AfterAll(func() {
+		// Find all AWS resources created in tests
+		var tagFilters []*resourcegroupstaggingapi.TagFilter
+		for key, value := range k8sTestTags {
+			tagFilters = append(tagFilters, &resourcegroupstaggingapi.TagFilter{
+				Key:    aws.String(key),
+				Values: []*string{value},
+			})
+		}
+		getResourcesInput := &resourcegroupstaggingapi.GetResourcesInput{
+			TagFilters: tagFilters,
+		}
+
+		deleteTaggedResources := func(
+			getResourcesResult *resourcegroupstaggingapi.GetResourcesOutput,
+			vpcLatticeClient *vpclattice.VPCLattice,
+			ramClient *ram.RAM,
+		) {
+			for _, mapping := range getResourcesResult.ResourceTagMappingList {
+				arn, err := arn2.Parse(*mapping.ResourceARN)
+				Expect(err).ToNot(HaveOccurred())
+
+				resourceType := strings.Split(arn.Resource, "/")[0]
+				switch resourceType {
+				case "servicenetwork":
+					deleteSnInput := &vpclattice.DeleteServiceNetworkInput{
+						ServiceNetworkIdentifier: aws.String(arn.String()),
+					}
+					_, err := vpcLatticeClient.DeleteServiceNetworkWithContext(ctx, deleteSnInput)
+					Expect(err).ToNot(HaveOccurred())
+				case "resource-share":
+					deleteShareInput := &ram.DeleteResourceShareInput{
+						ResourceShareArn: aws.String(arn.String()),
+					}
+					_, err := ramClient.DeleteResourceShareWithContext(ctx, deleteShareInput)
+					Expect(err).ToNot(HaveOccurred())
+				default:
+					Fail(fmt.Sprintf("Unknown resource type %s", resourceType))
+				}
+			}
+		}
+
+		// Delete secondary account AWS resources
+		secondaryGetResourcesResult, err := secondaryResourceGroupsClient.GetResources(getResourcesInput)
+		Expect(err).ToNot(HaveOccurred())
+		deleteTaggedResources(secondaryGetResourcesResult, secondaryVpcLatticeClient, secondaryRamClient)
+
+		// Delete primary account AWS resources
+		primaryGetResourcesResult, err := primaryResourceGroupsClient.GetResources(getResourcesInput)
+		Expect(err).ToNot(HaveOccurred())
+		deleteTaggedResources(primaryGetResourcesResult, primaryVpcLatticeClient, primaryRamClient)
+	})
+})
+
+func createSharedServiceNetwork(
+	serviceNetworkName string,
+	secondaryAccountId string,
+	primaryVpcLatticeClient *vpclattice.VPCLattice,
+	secondaryVpcLatticeClient *vpclattice.VPCLattice,
+	primaryRamClient *ram.RAM,
+	secondaryRamClient *ram.RAM,
+) *vpclattice.GetServiceNetworkOutput {
```

Comment:
you can create struct that will hold all clients and create smaller methods that will create and delete resources

Reply:
Done next commit!